Message send function

kawa18sima opened the pull request:
# WHAT
Make message function
# WHY
Because it is necessary to post a message

Commit:
Create message model

Using "rails g model message" command

Commit:
Create the messages table

Create the messages table
Using "rake db: migrate" command

Commit:
Define association

Message Define association of model relation

Commit:
Create group index

Change root view
Move partial template to common folder

Commit:
Add routing for messages create

Add routing of create action of messages

Commit:
Edit messages controller

Index action
· Group group information on the page
· Group to which the user belongs
When sending a message
· Group message
Add instance variable

Create action
Save processing of transmission data
Redirect processing to index
Add

Commit:
Edit messages index view

Message index view
Change group name, members, and messages to be displayed
Changed to be able to send messages
Add link for group edit

Change display part of message to partial template

Commit:
Edit group bar on the right side

Display the final message within a group

kawa18sima commented:
バリデーションとflashメッセージは次のブランチで実装します
viewはこんな感じです。
<img width="1440" alt="2017-05-09 1 12 38" src="https://cloud.githubusercontent.com/assets/24731933/25813621/36150dee-3455-11e7-8dee-71cfbb7a1e7a.png">

kinshotomoya commented on app/views/messages/index.html.haml:
```diff
@@ -8,29 +8,22 @@
+        =@group.name
+      %p
+        Members:
+        -@group.users.each do |member|
```
@group.usersにすると、viewでもSQL回ってしまうので、
コントローラで定義しよう！

kinshotomoya replied:
ここ、修正した？？

kinshotomoya commented on app/views/messages/index.html.haml:
```diff
@@ -8,29 +8,22 @@
+        Members:
+        -@group.users.each do |member|
+          ="#{member.name}"
+    = link_to "Edit",edit_group_path(params[:group_id]) ,class: "group-space__edit"
```
ここも！
コントローラでparamsの値を取ってきたら、直接paramsから取ってくる処理をしなくていいようになるから、
コントローラ側を考えよう！

kinshotomoya replied:
ここも！

kinshotomoya commented on app/views/common/_group-lists.html.haml:
```diff
@@ -2,4 +2,8 @@
       =group.name
-    %p まだメッセージはありません
+    %p
+      - if group.messages[0]
```
これでいけた？？

kawa18sima replied:
すみません、できていませんでした。。。

kinshotomoya reviewed (commented):
コメントしました！！！！！！！
下記修正してください！

Commit:
Edit messages index and controller

Group.users.each part
To controller
@group_members = group.users
Define and replace

Params [: user_id] that was in index is
Change to @ group.id

Commit:
Edit _group-lists.html.haml

Because the last message of group was not displayed

group.messages [0] -> group.messages.last

kawa18sima commented:
編集しました！
確認不足ですみません。
<img width="1439" alt="2017-05-10 8 10 58" src="https://cloud.githubusercontent.com/assets/24731933/25876447/4986da48-3558-11e7-839c-3a3876c08bb5.png">

kinshotomoya reviewed (commented):
コメントしました！
遅くなってごめん！

kinshotomoya commented:
おっけです！！！！！！
いい感じ！！
マージして次行きましょう！！！！！！！！！
